File: apps/ios/AceClub/Domain/Entities/Invitation.swift
import Foundation

struct Invitation: Identifiable {
    let id: String
    let organizationId: String
    let email: String
    let role: MemberRole
    let status: InvitationStatus
    let expiresAt: String
    let createdAt: String?
    let inviterId: String
    let organization: Organization?

    var isExpired: Bool {
        let formatter = ISO8601DateFormatter()
        formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        guard let expireDate = formatter.date(from: expiresAt) else {
            // Try without fractional seconds
            formatter.formatOptions = [.withInternetDateTime]
            guard let expireDate = formatter.date(from: expiresAt) else {
                return false
            }
            return expireDate < Date()
        }
        return expireDate < Date()
    }

    var isPending: Bool { status == .pending }
}

enum InvitationStatus: String, Codable {
    case pending
    case accepted
    case rejected
    case canceled
}

File: apps/mobile/types/invitation.ts
import type { MemberRole } from "./common";
import type { Organization } from "./organization";

export type InvitationStatus = "pending" | "accepted" | "rejected" | "canceled";

export interface Invitation {
  id: string;
  organizationId: string;
  email: string;
  role: MemberRole;
  status: InvitationStatus;
  expiresAt: string;
  createdAt?: string | null;
  inviterId: string;
  organizationName?: string | null;
  organization?: Organization | null;
}

export interface ListInvitationsResponse {
  invitations: Invitation[];
}

export interface CreateInvitationRequest {
  email: string;
  role: MemberRole;
  organizationId: string;
}

export interface AcceptInvitationResponse {
  member: {
    id: string;
    organizationId: string;
    userId: string;
    role: string;
    createdAt: string;
  };
  organization: Organization;
}
